strawberryfields/io.py

```diff
@@ -81,7 +83,29 @@ def to_blackbird(prog, version="1.0"):
                     a = str(a)
                 op["args"].append(a)
 
+        # if program type is "tdm" then add the looped-over arrays to the blackbird program
+        if prog.type == "tdm":
+            for p in prog.loop_vars:
+                for i, ar in enumerate(op["args"]):
+                    if str(p) == str(ar):
+                        op["args"][i] = p.name
+                for k, v in op["kwargs"].items():
+                    if str(p) == str(v):
+                        op["kwargs"][k] = p.name
+
         bb._operations.append(op)
+    # add the specific "tdm" metadata to the Blackbird program
+    if prog.type == "tdm":
+        bb._type["name"] = "tdm"
+        bb._type["options"].update(
+            {
+                "temporal_modes": prog.timebins,
```

Comment:
Might be worth ironing out this one across the components (either going with `temporal modes` in SF too or having `timebins` elsewhere). Any thoughts?

Reply:
Thought about this as well. Not sure what the preference would be. I started using `temporal modes` and then later noticed that `timebins` was used inside `TDMProgram`. I don't necessarily mind using both, but I agree it might be a bit confusing having two different names for the same thing. I guess it depends on how commonly used these terms are outside of our software libraries. Any inputs @nquesada?

Reply:
Also @fab1an-q!

Reply:
It might be subtle but the difference between timebins and temporal modes occurs when we have more than one spatial mode where within one timebin we may deal with many temporal qumodes (think of the Tokyo and DTU experiments). This is why I prefer timebins, personally.